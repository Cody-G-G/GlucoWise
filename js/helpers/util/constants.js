'use strict';

const readingUnitStandards = {
    UK: 'mmol/L',
    US: 'mg/dL'
};

const dataModes = {
    glucose: "Glucose",
    steps: "Steps",
    calories: "Calories",
    weight: "Weight"
};

const timeRanges = {
    lastHour: "60m",
    lastDay: "24h",
    lastWeek: "7d",
    lastMonth: "30d",
    lastHalfYear: "6M",
    lastYear: "1y"
};

const defaultSafeRange = {
    min: '70',
    max: '130'
};

const dbObjects = {
    reading: "BGLReading",
    safeRange: "BGLSafeRange",
    standard: "BGLStandard",
    dataSync: "DataSyncSettings",
    foodItem: "ConsumedFoodItem"
};

const daysOfWeek = ["Sunday", "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday"];
const monthsOfYear = ["January", "February", "March", "April", "May", "June", "July", "August", "September", "Octomber", "November", "December"];

export {readingUnitStandards, dataModes, timeRanges, defaultSafeRange, daysOfWeek, monthsOfYear, dbObjects};