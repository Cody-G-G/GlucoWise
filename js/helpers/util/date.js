'use strict';
import moment from "moment";
const dayOfWeek = ["Sunday", "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday"];

export default {
    /**
     * @param datePortion
     * @returns {string}
     */
    padDate(datePortion) {
        return datePortion < 10 ? "0" + datePortion : datePortion;
    },

    getTodayString() {
        return this.toDateString(new Date());
    },

    /**
     * @param date
     * @returns {string}
     */
    toDateString(date) {
        return this.padDate(date.getDate()) + "-" + this.padDate(date.getMonth() + 1) + "-" + date.getFullYear();
    },

    /**
     * @param date
     * @returns {string}
     */
    toDateTimeString(date) {
        return this.toDateString(date) + "  " + this.padDate(date.getHours()) + ":" + this.padDate(date.getMinutes());
    },

    /**
     * @param from
     * @param to
     * @returns {boolean}
     */
    isWithin24Hours(from, to) {
        return (Math.abs(from - to) / 36e5) <= 24;
    },

    /**
     * @returns {boolean}
     * @param from
     * @param to
     */
    areWithin60Minutes(from, to) {
        return (Math.abs(from - to) / 36e5) <= 1;
    },

    /**
     * @param date
     * @returns {string}
     */
    dayOfWeek(date){
        return dayOfWeek[new Date(date).getDay()];
    },

    /**
     * @param from
     * @param to
     * @returns {number}
     */
    hoursBetween(from, to) {
        return Math.abs(from - to) / 36e5;
    },

    /**
     * @param from
     * @param to
     * @returns {number}
     */
    daysBetween(from, to) {
        return Math.abs(from - to) / 864e5;
    },

    /**
     * @param from
     * @param to
     * @returns {number}
     */
    minutesBetween(from, to) {
        return Math.abs(from - to) / 6e4;
    },

    /**
     * @param from
     * @param to
     * @returns {number}
     */
    weeksBetween(from, to) {
        return Math.abs(from - to) / 6048e5;
    },

    /**
     * @param from
     * @param to
     * @returns {number}
     */
    monthsBetween(from, to) {
        return Math.abs(from - to) / 2628e6;
    },

    /**
     * @param dateString
     * @param hour
     * @param minute
     * @param second
     * @param millisecond
     * @returns {Date}
     */
    toDateFromDateString(dateString, hour, minute, second, millisecond) {
        return new Date(moment(dateString, "DD-MM-YYYY").hour(hour).minute(minute).second(second).millisecond(millisecond));
    },

    /**
     * @param dateTimeString
     * @returns {Date}
     */
    toDateFromDateTimeString(dateTimeString) {
        return new Date(moment(dateTimeString, "DD-MM-YYYY HH:mm"));
    },

    /**
     * @param hours
     * @param date
     * @returns {number}
     */
    hoursAgoMillis(hours, date) {
        return new Date(moment(date).subtract(hours, 'hours')).getTime();
    },

    /**
     * @param days
     * @param date
     * @returns {number}
     */
    daysBeforeMillis(days, date) {
        return new Date(moment(date).subtract(days, 'days')).getTime();
    },

    /**
     * @param months
     * @param date
     * @returns {number}
     */
    monthsAgoMillis(months, date) {
        return new Date(moment(date).subtract(months, 'months')).getTime();
    },

    dayStartMillis(date) {
        return new Date(moment(date).startOf('day')).getTime();
    },

    dayEndMillis(date) {
        return new Date(moment(date).endOf('day')).getTime();
    },

    hourStartMillis(date) {
        return new Date(moment(date).startOf('hour')).getTime();
    },

    minuteStartMillis(date) {
        return new Date(moment(date).startOf('minute')).getTime();
    },

    millisFromMidnight(date) {
        return date - this.dayStartMillis(date);
    },

    millisFromHourStart(date) {
        return date - this.hourStartMillis(date);
    },

    millisFromMinuteStart(date) {
        return date - this.minuteStartMillis(date);
    },

    /**
     * @param hoursAgo
     * @param date
     * @returns {number}
     */
    hourOfDayHoursAgo(hoursAgo, date) {
        return new Date(this.hoursAgoMillis(hoursAgo, date)).getHours();
    }
};