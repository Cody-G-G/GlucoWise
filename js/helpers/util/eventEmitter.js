'use strict';
const EventEmitter = require('events');
const eventEmitter = new EventEmitter();

class Emitter {

    emitGraphsHelpEvent() {
        eventEmitter.emit('GraphsHelp');
    }

    addGraphsHelpListener(callback) {
        eventEmitter.addListener('GraphsHelp', callback);
    }
}

export default new Emitter();
