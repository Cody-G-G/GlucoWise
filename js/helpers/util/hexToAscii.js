'use strict';
export default (hexArg) => {
    const hexArgStr = hexArg.toString();
    let resultStr = '';
    for (let i = 0; i < hexArgStr.length; i += 2)
        resultStr += String.fromCharCode(parseInt(hexArgStr.substr(i, 2), 16));
    return resultStr;
}