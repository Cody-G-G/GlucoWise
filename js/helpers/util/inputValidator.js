'use strict';

const isNumberValid = (input) => {
    let numbers = '0123456789.';
    let decimalPoints = 0;
    let valid = true;
    let digits = String(input).trim().split('');

    let isInputEmpty = digits.length === 0;
    let isLastCharDecimalPoint = digits[digits.length - 1] === '.';
    let isZero = input === 0;

    digits.forEach((digit) => {
        let isValidDigit = (numbers.indexOf(digit) !== -1);
        digit === '.' && decimalPoints++;

        if (!isValidDigit || decimalPoints > 1) {
            valid = false;
        }
    });

    (isInputEmpty || isLastCharDecimalPoint || isZero) && (valid = false);

    return valid;
};

export default isNumberValid;