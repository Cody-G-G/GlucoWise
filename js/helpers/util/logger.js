'use strict';
export default (string) => {
    console.log(string);
}