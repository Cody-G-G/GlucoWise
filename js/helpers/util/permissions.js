'use strict';
import log from "./logger";
import {PermissionsAndroid} from 'react-native';
import LocationServicesDialogBox from "react-native-android-location-services-dialog-box";

export default {
    requestLocationServices: () => {
        log("Requesting Location Services");
        return new Promise((resolve, reject) => {
            LocationServicesDialogBox.checkLocationServicesIsEnabled({
                message: "<h2>Use Location ?</h2>This app wants to change your device settings:<br/><br/>Use GPS, Wi-Fi, and cell network for location<br/><br/><a href='#'>Learn more</a>",
                ok: "YES",
                cancel: "NO"
            })
                .then((success) => {
                    log("Location Service Request Success: " + success);
                    resolve();
                })
                .catch((error) => {
                    log("Location Service Request ERROR: " + error.message);
                })
        })
    },

    requestLocationCoarsePermission: () => {
        log("Requesting Location Coarse Permission");
        return new Promise((resolve, reject) => {
            PermissionsAndroid.check(PermissionsAndroid.PERMISSIONS.ACCESS_COARSE_LOCATION).then((alreadyGranted) => {
                alreadyGranted && resolve(true);
                PermissionsAndroid.request(PermissionsAndroid.PERMISSIONS.ACCESS_COARSE_LOCATION, {
                    'title': 'Location permission',
                    'message': 'In order to use bluetooth, the app needs Location permissions. To enable, tap the screen outside of this box, after which you will be prompted.'
                })
                    .then((result) => {
                        const granted = result === PermissionsAndroid.RESULTS.GRANTED;
                        log("Permission LOCATION_COARSE granted: " + granted);
                        resolve(granted);
                    })
                    .catch((error) => {
                        log("Location Coarse Permission ERROR: " + error);
                    });
            })
        })
    }
};