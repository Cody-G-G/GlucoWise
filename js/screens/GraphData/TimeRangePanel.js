'use strict';
import React, {Component} from'react';
import {View} from 'react-native';
import {timeRanges} from "../../helpers/util/constants";
import ToggleButton from "../../helpers/components/ToggleButton";
import styles from "./styles";

export default class TimeRangePanel extends Component {
    constructor(props) {
        super(props);
    }

    render() {
        return (
            <View style={styles.timeRangePanel}>
                <ToggleButton type={timeRanges.lastDay}
                              selectedType={this.props.timeRange}
                              onPress={() => this.props.onPress(timeRanges.lastDay)}
                              onColor='royalblue'
                              offColor='darkgrey'
                              onText={timeRanges.lastDay}
                              offText={timeRanges.lastDay}/>
                <ToggleButton type={timeRanges.lastHour}
                              selectedType={this.props.timeRange}
                              onPress={() => this.props.onPress(timeRanges.lastHour)}
                              onColor='royalblue'
                              offColor='darkgrey'
                              onText={timeRanges.lastHour}
                              offText={timeRanges.lastHour}/>
            </View>
        );
    }
}