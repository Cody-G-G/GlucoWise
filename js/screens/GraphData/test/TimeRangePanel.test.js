'use strict';
import React from 'react';
import 'react-native';
import renderer from 'react-test-renderer';
import TimeRangePanel from "../TimeRangePanel";
import {timeRanges} from "../../../helpers/util/constants";


test('TimeRangePanel - renders correctly', () => {
    let timeRange = timeRanges.lastHour;

    const tree = renderer.create(
        <TimeRangePanel timeRange={timeRange}/>
    ).toJSON();

    expect(tree).toMatchSnapshot();
});