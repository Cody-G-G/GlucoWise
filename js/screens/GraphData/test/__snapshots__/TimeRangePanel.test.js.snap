// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimeRangePanel - renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "stretch",
      "flex": 1,
      "flexDirection": "row",
    }
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "darkgrey",
        "borderWidth": 1,
        "flex": 1,
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "white",
          "fontSize": 35,
          "fontWeight": "bold",
          "padding": 5,
        }
      }
    >
      24h
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "royalblue",
        "borderWidth": 1,
        "flex": 1,
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "white",
          "fontSize": 35,
          "fontWeight": "bold",
          "padding": 5,
        }
      }
    >
      60m
    </Text>
  </View>
</View>
`;
